packages/playwright/src/reporters/list.ts

```diff
@@ -80,17 +80,17 @@ class ListReporter extends BaseReporter {
     if (step.category !== 'test.step')
       return;
     const testIndex = this._resultIndex.get(result) || '';
+    const ordinal = ((result as any)[lastStepOrdinalSymbol] || 0) + 1;
```

Comment:
these lines were just shifted up before the `return`, so that the `stepIndex` is filled.